assets/src/editor/blocks/donation-bar/index.js

```diff
@@ -0,0 +1,254 @@
+							min="1"
+							value={ defaultAmount }
+							onChange={ ( value ) => setAttributes( { defaultAmount: value } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Annual Option', 'shiro' ) }
+							checked={ showAnnual }
+							onChange={ () => setAttributes( { showAnnual: ! showAnnual } ) }
+						/>
+						<TextControl
+							label={ __( 'UTM Medium', 'shiro' ) }
+							value={ medium }
+							onChange={ ( value ) => setAttributes( { medium: value } ) }
+						/>
+						<TextControl
+							label={ __( 'UTM Campaign', 'shiro' ) }
+							value={ campaign }
+							onChange={ ( value ) => setAttributes( { campaign: value } ) }
+						/>
+						<TextControl
+							label={ __( 'UTM Source', 'shiro' ) }
+							value={ source }
+							onChange={ ( value ) => setAttributes( { source: value } ) }
+						/>
+					</PanelBody>
+					<PanelBody title={ __( 'Default Amounts by Currency', 'shiro' ) } initialOpen={ false }>
+						{ defaultCurrencyAmount.map( ( item, idx ) => {
+							const usedCurrencies = defaultCurrencyAmount
+								.filter( ( _, i ) => i !== idx )
+								.map( ( entry ) => entry.currency );
+							const availableCurrencies = supportedCurrencies.filter(
+								(opt) => ! usedCurrencies.includes( opt.value ) || opt.value === item.currency
+							);
+							return (
+								<Flex key={ idx } align="center" style={ { marginBottom: 8 } }>
+									<FlexItem>
+										<SelectControl
+											label={ idx === 0 ? __( 'Currency', 'shiro' ) : '' }
+											value={ item.currency }
+											options={ availableCurrencies }
+											onChange={ ( value ) => updateCurrencyAmount( idx, 'currency', value ) }
+											style={ { width: 120 } }
+										/>
+									</FlexItem>
+									<FlexItem>
+										<TextControl
+											label={ idx === 0 ? __( 'Amount', 'shiro' ) : '' }
+											type="number"
+											min="1"
+											value={ item.amount }
+											onChange={ ( value ) => updateCurrencyAmount( idx, 'amount', Number( value ) ) }
+											placeholder="10"
+											style={ { width: 80 } }
+										/>
+									</FlexItem>
+									<FlexItem>
+										<Button
+											isDestructive
+											icon="no-alt"
+											label={ __( 'Remove', 'shiro' ) }
+											onClick={ () => removeCurrencyAmount( idx ) }
+										/>
+									</FlexItem>
+								</Flex>
+							);
+						} ) }
+						<Button
+							variant="secondary"
+							onClick={ addCurrencyAmount }
+							disabled={ defaultCurrencyAmount.length >= supportedCurrencies.length }
+						>
+							{ __( 'Add Currency Amount', 'shiro' ) }
+						</Button>
+					</PanelBody>
+				</InspectorControls>
+				<div { ...blockProps }>
+					<ServerSideRender
+						block={ metadata.name }
+						attributes={ attributes }
+					/>
+				</div>
+			</>
+		);
+	},
+
+	/**
+	 * Save the share article block, it's a dynamic block.
+	 */
+	save: function Save( { attributes } ) {
+		return null;
+	},
```

Comment:
If the save method is only to return null, we can omit the save method entirely during registration and it'll naturally fall back to the PHP render